src/core/PathUtil.cpp

```diff
@@ -125,39 +141,50 @@ namespace lmms::PathUtil
 
 	QString oldRelativeUpgrade(const QString & input)
 	{
-		if (input.isEmpty()) { return input; }
+		Base assumedBase = Base::Absolute;
+
+		if (input.isEmpty()) { return basePrefix(assumedBase); }
+
+		QString path = serializePath(input);
 
-		//Start by assuming that the file is a user sample
-		Base assumedBase = Base::UserSample;
+		//Check if it's a user sample.
+		QString userPath = baseLocation(Base::UserSample) + path;
+		QFileInfo userInfo(userPath);
+		if (userInfo.exists()) { assumedBase = Base::UserSample; }
 
 		//Check if it's a factory sample
-		QString factoryPath = baseLocation(Base::FactorySample) + input;
+		QString factoryPath = baseLocation(Base::FactorySample) + path;
 		QFileInfo factoryInfo(factoryPath);
 		if (factoryInfo.exists()) { assumedBase = Base::FactorySample; }
 
 		//Check if it's a VST
-		QString vstPath = baseLocation(Base::UserVST) + input;
+		QString vstPath = baseLocation(Base::UserVST) + path;
 		QFileInfo vstInfo(vstPath);
 		if (vstInfo.exists()) { assumedBase = Base::UserVST; }
 
 		//Assume we've found the correct base location, return the full path
-		return basePrefix(assumedBase) + input;
+		return basePrefix(assumedBase) + path;
 	}
 
 
 
 
 	QString toAbsolute(const QString & input, bool* error /* = nullptr*/)
 	{
+		if (input.isEmpty()) { return input; }
+
+		QString path = serializePath(input);
+
 		//First, do no harm to absolute paths
-		QFileInfo inputFileInfo = QFileInfo(input);
+		// Note: Paths starting with a colon (:) are always considered absolute, as they denote a QResource.
+		QFileInfo inputFileInfo = QFileInfo(path);
 		if (inputFileInfo.isAbsolute())
 		{
 			if (error) { *error = false; }
-			return input;
+			return path;
 		}
 		//Next, handle old relative paths with no prefix
-		QString upgraded = input.contains(":") ? input : oldRelativeUpgrade(input);
+		QString upgraded = path.contains(":") ? path : oldRelativeUpgrade(path);
 
 		Base base = baseLookup(upgraded);
```

Comment:
Since ...

> Paths starting with a colon (:) are always considered absolute, as they denote a QResource.

... this check is now redundant.  I've proposed a simplified version as a code suggestion.


```suggestion
		Base base = baseLookup(oldRelativeUpgrade(path));
```

Reply:
Yeah, oversimplified.  Proposed a new patch, resolving this outdated convo.